### PR TITLE
Add a show-paper-key path to provisioning flow

### PR DESCRIPTION
**Pull request by cjb**
@keybase/react-hackers 

There's a (fairly common) provisioning path that was missing the show-paper-key stage -- when you have a keybase account with no device keys and login through the GUI app.  Before this PR, the `displayPrimaryPaperKey` RPC is unhandled and we just move to being logged in without ever showing the user their new paper key.

After this PR, we go to the same show-paper-key screen that signup uses as its Success page, but now parameterized to say appropriate things for our case.

I'm not going to merge this until I see whether this breaks the signup flow; I can test that once DESKTOP-818 is fixed.

### CODE REVIEW

File: shared/login/signup/success/index.js [hunk 1]
```diff
@@ -25,9 +25,10 @@ Success.propTypes = {
 }
 
 export default connect(
-  state => ({paperkey: state.signup.paperkey}),
+  (state, ownProps) => ({
+    paperkey: ownProps.paperkey || state.signup.paperkey,
```
Comment (cjb) [inline shared/login/signup/success/index.js]: This is hacky -- we can either be being called from redux with entries in the store (which the signup flow prefers), or directly with `appendRouteElement()` (which the login flow prefers), so we try to support both.

File: shared/login/signup/success/index.js [hunk 2]
```diff
@@ -25,9 +25,10 @@ Success.propTypes = {
...
+    title: ownProps.title || state.signup.successTitle
+  }),
   dispatch => ({
-    onFinish: () => dispatch(sawPaperKey()),
```
Comment (cjb) [inline shared/login/signup/success/index.js]: This'll probably break signup.  I wonder if there's a way to do something similar to the `(state, ownProps)` above but for dispatch.  We want to dispatch login's `sawPaperKey()` if we're in the login flow and signup's `sawPaperKey()` if we're in the signup flow.

Comment (chrisnojima) [inline shared/login/signup/success/index.js]: dispatch takes own props also

Comment (chrisnojima) [inline shared/login/signup/success/index.js]: By dispatch I mean this closure takes dispatch and own props

File: shared/actions/login/index.js [hunk 3]
```diff
@@ -467,6 +479,16 @@ function makeKex2IncomingMap (dispatch, getState) : incomingCallMapType {
...
           onBack={() => dispatch(cancelLogin(response))} />))
     },
+    'keybase.1.loginUi.displayPrimaryPaperKey': ({sessionID, phrase}, response) => {
+      paperKeyResponse = response
+      let hiddenphrase = new HiddenString(phrase)
+      appendRouteElement((
+        <Success
+          paperkey={hiddenphrase}
+          onFinish={() => dispatch(sawPaperKey())}
+          onBack={() => dispatch(cancelLogin(response))}
```
Comment (chrisnojima) [inline shared/actions/login/index.js]: You can just resolve the response sw here. You don't need to store it above

File: shared/login/signup/success/index.render.desktop.js [hunk 4]
```diff
@@ -23,7 +23,7 @@ export default class Render extends Component {
   render () {
     return (
       <Container onBack={this.props.onBack} style={styles.container}>
-        <Text type='Header' style={styles.header}>Congratulations, you’ve just joined Keybase!</Text>
+        <Text type='Header' style={styles.header}>{this.props.title ? this.props.title : 'Congratulations, you’ve just joined Keybase!'}</Text>
```
Comment (MarcoPolo) [inline shared/login/signup/success/index.render.desktop.js]: :sheep: `this.props.title || 'Congratulations...'`

File: shared/login/signup/success/index.js [hunk 5]
```diff
@@ -25,9 +25,10 @@ Success.propTypes = {
 }
 
 export default connect(
-  state => ({paperkey: state.signup.paperkey}),
+  (state, ownProps) => ({
+    paperkey: ownProps.paperkey || state.signup.paperkey,
+    title: ownProps.title || state.signup.successTitle
```
Comment (MarcoPolo) [inline shared/login/signup/success/index.js]: Where did this `successTitle` come from?

**Comment by MarcoPolo** [comment]
What if instead of using the Smart Component for the Success page, you just use the dumb Success render component. I think that has everything you need and you won't need to mess with the `connect`stuff.

**Comment by cjb** [comment]
Thanks, followed all of the review comments!  There's way less changed code now.  Another look for thumbs?

**Comment by chrisnojima** [comment]
looking

**Comment by chrisnojima** [comment]
:+1:

**Pull request merged**